src/model/FanComponentModel.hpp

```diff
@@ -0,0 +1,296 @@
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_FANCOMPONENTMODEL_HPP
+#define MODEL_FANCOMPONENTMODEL_HPP
+
+#include <model/ModelAPI.hpp>
+#include "StraightComponent.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+  class Curve;
+  class Schedule;
+  class AirflowNetworkFan;
+
+  namespace detail {
+
+    class FanComponentModel_Impl;
+
+  }  // namespace detail
+
+  /** FanComponentModel is a StraightComponent that wraps the OpenStudio IDD object 'OS:Fan:ComponentModel'. */
+  class MODEL_API FanComponentModel : public StraightComponent
+  {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    // Constructor that will instantiate all required curves and optional curves too
+    explicit FanComponentModel(const Model& model);
+
+    // Explicit constructor that takes in all required curves. It will **not** instantiate the optional curve objects.
+    // You can then call the helper method `bool assignDefaultOptionalCurves()` if that is what you want
+    explicit FanComponentModel(const Model& model, const Curve& fanPressureRiseCurve, const Curve& ductStaticPressureResetCurve,
+                               const Curve& normalizedFanStaticEfficiencyCurveNonStallRegion,
+                               const Curve& normalizedFanStaticEfficiencyCurveStallRegion,
+                               const Curve& normalizedDimensionlessAirflowCurveNonStallRegion,
+                               const Curve& normalizedDimensionlessAirflowCurveStallRegion);
```

Comment:
@kbenne these are my two new Ctors. As you can see, the default one also defaults the optional Curves... perhaps that should be changed?

The other one that takes required curve object does not. I assume user is trying to avoid duplicated/unnecessary curves by using this more complicated ctor.

Reply:
In the first ctor, are the optional curves required to run a simulation by EnergyPlus? If they are then 1. I wonder if they should really be optional and 2. I think you did the right thing by instantiating them.

If however, these optional curves are not needed to run a simulation then I think the ctor should not default construct them.

Reply:
They aren't required. According to the I/O ref: All of the optional curves: if blank then use 1.0 instead. Except VRF Efficency Curve Name, if blank = 0.97.

see https://bigladdersoftware.com/epx/docs/9-4/input-output-reference/group-fans.html#field-maximum-belt-efficiency-curve-name